src/webgl/p5.RendererGL.js

```diff
@@ -32,77 +33,134 @@ defineStrokeJoinEnum('ROUND', 0);
 defineStrokeJoinEnum('MITER', 1);
 defineStrokeJoinEnum('BEVEL', 2);
 
-const lightingShader = readFileSync(
+p5.RendererGL.prototype.getCachedShader = function (
+  shaderKey, vertexShaderSource, fragmentShaderSource) {
+  if (!this._pInst.shaderCache) {
+    this._pInst.shaderCache = {};    //ensures a unique shaderCache for each instance of p5.RendererGL
+  }
+  if (!this._pInst.shaderCache[shaderKey]) {
+    const vertexShader = this._pInst.createShader(
+      this._pInst.VERTEX_SHADER, vertexShaderSource);
+    const fragmentShader = this._pInst.createShader(
+      this._pInst.FRAGMENT_SHADER, fragmentShaderSource);
+
+    this._pInst.shaderCache[shaderKey] = this._pInst.createShaderProgram(
+      vertexShader, fragmentShader);
+  }
+  return this._pInst.shaderCache[shaderKey];
+};
+
+const lightingShader = p5.RendererGL.prototype.getCachedShader('lightingShader',readFileSync(
```

Comment:
I think right now this is creating `p5.Shader` instances where before it was just loading shader *source code.* We still need to load the source code here, but we can't actually instantiate a `p5.Shader` until we have a p5 *instance*. Right now, you're calling the method directly on the class itself rather than an instance of the class, so `this` will be undefined.

I think the spot to call the `getCachedShader` method is inside of instance method like `getDefaultFillShader()`.